dev-db/postgresql/files/postgresql.init-9.3-r1

```diff
@@ -144,14 +173,32 @@ description_reload="Simply sends the postgres process a SIGHUP signal, causing
            require a complete restart to take effect."
 reload() {
     ebegin "Reloading PostgreSQL @SLOT@ configuration"
-    su - postgres -c "${PG_CTL} reload -s -D ${DATA_DIR}"
+    start-stop-daemon \
+        --signal SIGHUP \
+        --user postgres:postgres \
+        --pidfile "${PIDFILE}"
     eend $?
 }
 
 description_promote="If the server is in standby, it is commanded to exit
             recovery and begin read-write operations."
 promote() {
     ebegin "Promoting PostgreSQL @SLOT@"
-    su - postgres -c "${PG_CTL} promote -s -D ${DATA_DIR}"
-    eend $?
+    local retval=1
+    if "${PG_CONTROLDATA}" -D "${DATA_DIR}" 2>/dev/null \
+        | grep 'in archive recovery' >/dev/null 2>&1; then
+        local promotefile="${DATA_DIR%/}/promote"
+        checkpath -q -f -m 644 -o postgres:postgres "${promotefile}" && \
+            start-stop-daemon \
+                --signal SIGUSR1 \
+                --user postgres:postgres \
+                --pidfile "${PIDFILE}"
+        retval=$?
+        if [ ${retval} -ne 0 ] ; then
+            rm -f "${promotefile}"
+        fi
```

Comment:
Related to previous comment.

Reply:
As mentioned above, if the sending signal fails, the file will not be deleted by the postgresql daemon, so you need to consider the failure and delete it manually. This behavior is consistent with pg_ctl